### PR TITLE
Syntax changes for multiple Kafka metadata items

### PR DESCRIPTION
**Pull request by quodlibetor**
Part of the implementation of #1577 (design doc at #8239)

### Motivation

  * This PR works towards a known-desirable feature

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR adds a release note for any user-facing behavior changes.

### CODE REVIEW

File: src/dataflow-types/src/types.rs [hunk 1]
```diff
@@ -965,18 +980,30 @@ pub struct KafkaSourceConnector {
     // Map from partition -> starting offset
     pub start_offsets: HashMap<i32, i64>,
     pub group_id_prefix: Option<String>,
-    pub key_envelope: KeyEnvelope,
+    pub include_items: Vec<KafkaMetadataItem>,
```
[the remark this answers is not in the document]
Comment (quodlibetor) [reply src/dataflow-types/src/types.rs]: I'll handle that in a follow up PR though, it's not super relevant to these changes, and there's no reason to block @petrosagg's work on default-including metadata on reviewing it.

**Comment by quodlibetor** [comment]
@benesch when you get a chance could you re-review this?

**Review by benesch — APPROVED** [review]
Looks great to me! Sorry for the delay.

**Review by petrosagg — COMMENTED** [review]
Can we also add an `INCLUDE OFFSET` variant before merging this PR that will be used to enable `mz_offset` in a subsequent PR?

**Comment by quodlibetor** [comment]
yup, done @petrosagg

**Comment by quodlibetor** [comment]
hmm actually right now `INCLUDE OFFSET` is not technically unsupported, it's just the default.

**Comment by quodlibetor** [comment]
but we don't support renames or the future default name, so I think that this is still probably the best bet.

**Pull request merged**